### PR TITLE
Improves godoc

### LINKED ISSUE #356: GDoc client/interface.go
And add examples https://golang.org/src/sort/

### PR DESCRIPTION
**Pull request by chuckha**
Adds examples to client and results packages.

Closes #356

Signed-off-by: Chuck Ha <chuck@heptio.com>

### CODE REVIEW

File: pkg/client/results/doc.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package results provides a low level api to extract data out of a Sonobuoy
+// result archive.
```
Comment (Bradamant3) [inline pkg/client/results/doc.go]: s/api/API
s/out of/from

File: pkg/client/example_interfaces_test.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+			EnableRBAC:      true,
+			ImagePullPolicy: "Always",
+		},
+	}
+
+	// Runs sonobuoy on the cluster configured in $HOME/.kube/config.
+	if err = sonobuoy.Run(&runConfig); err != nil {
+		panic(err)
+	}
+}
```
Comment (timothysc) [inline pkg/client/example_interfaces_test.go]: Could probably expand the examples for the different uses.

File: pkg/client/results/reader.go [hunk 3]
```diff
@@ -49,18 +49,22 @@ const (
...
-	versionNine    = "v0.9"
-	versionTen     = "v0.10"
+	VersionEight   = "v0.8"
+	VersionNine    = "v0.9"
+	VersionTen     = "v0.10"
 )
 
 // Reader holds a reader and a version. It uses the version to know
-// where to find certain files within the archive stream.
+// where to find certain files within the archive.
```
Comment (Bradamant3) [inline pkg/client/results/reader.go]: "certain" is vague and potentially can cause doubt. Can we either identify specifically or else omit?

File: pkg/client/results/reader.go [hunk 4]
```diff
@@ -49,18 +49,22 @@ const (
...
 type Reader struct {
 	io.Reader
 	Version string
 }
 
+// NewReaderWithVersion will create a results.Reader that can interpret a
```
Comment (Bradamant3) [inline pkg/client/results/reader.go]: s/will create/creates
s/can interpret/interprets

File: pkg/client/results/reader.go [hunk 5]
```diff
@@ -49,18 +49,22 @@ const (
...
+// results archive of the version passed in.
+// This is useful when the reader can only be read once and the version of the
```
Comment (Bradamant3) [inline pkg/client/results/reader.go]: Copyedit: "Useful if the reader can be read only once and if the version of the data to read is known."

File: pkg/client/results/reader.go [hunk 6]
```diff
@@ -212,55 +216,59 @@ func ExtractFileIntoStruct(file, path string, info os.FileInfo, object interface
...
+	case VersionEight:
 		return "serverversion/serverversion.json"
 	default:
 		return defaultServerVersionFile
 	}
 }
 
-// NamespacedResources returns the path to
-func (a *Reader) NamespacedResources() string {
+// NamespacedResources returns the path to the directory containing information
```
Comment (Bradamant3) [inline pkg/client/results/reader.go]: s/containing/that contains

File: pkg/client/results/reader.go [hunk 7]
```diff
@@ -212,55 +216,59 @@ func ExtractFileIntoStruct(file, path string, info os.FileInfo, object interface
...
 		return "resources/non-ns/"
 	default:
 		return nonNamespacedResourcesDir
 	}
 }
 
-// NodesFile returns the path to the file listing nodes.
-func (a *Reader) NodesFile() string {
-	return filepath.Join(a.NonNamespacedResources(), defaultNodesFile)
+// NodesFile returns the path to the file listing nodes of the Kubernetes cluster.
```
Comment (Bradamant3) [inline pkg/client/results/reader.go]: s/listing/that lists the

File: pkg/client/results/reader.go [hunk 8]
```diff
@@ -212,55 +216,59 @@ func ExtractFileIntoStruct(file, path string, info os.FileInfo, object interface
...
+func (r *Reader) NodesFile() string {
+	return filepath.Join(r.NonNamespacedResources(), defaultNodesFile)
 }
 
-// ServerGroupsFile returns the path to the groups the k8s api supported at the time of the run.
-func (a *Reader) ServerGroupsFile() string {
+// ServerGroupsFile returns the path to the groups the Kubernetes api supported at the time of the run.
```
Comment (Bradamant3) [inline pkg/client/results/reader.go]: s/api/API

File: pkg/client/results/types.go [hunk 9]
```diff
@@ -21,11 +21,13 @@ import (
 	"github.com/heptio/sonobuoy/pkg/discovery"
 )
 
-// Metadata is all the stuff about the Sonobuoy run and how long it took to query the system.
+// Metadata is the data about the Sonobuoy run itself and how long it took to
```
Comment (Bradamant3) [inline pkg/client/results/types.go]: omit "itself"

**Comment by chuckha** [comment]
thanks for the edits @Bradamant3 !

**Pull request merged**